usage/src/main/java/com/cloud/usage/UsageManagerImpl.java

```diff
@@ -275,7 +277,14 @@ public boolean configure(String name, Map<String, Object> params) throws Configu
             s_logger.error("Unhandled exception configuring UsageManger", e);
             throw new ConfigurationException("Unhandled exception configuring UsageManager " + e.toString());
         }
-        _pid = Integer.parseInt(System.getProperty("pid"));
+
+        try {
+            String processName = ManagementFactory.getRuntimeMXBean().getName();
+            _pid = Integer.parseInt(processName.split("@")[0]);
+        } catch (Exception e) {
+            s_logger.error("Unable to get process pid ", e);
+            throw new ConfigurationException("Unable to get process pid " + e.toString());
+        }
```

Comment:
1. twice the same message should be in a string constand
1. rethrown should not abandon the stack trace for the original problem
```suggestion
        String processName;
        try {
            processName = ManagementFactory.getRuntimeMXBean().getName();
            _pid = Integer.parseInt(processName.split("@")[0]);
        } catch (Exception e) {
            String msg = String.format("Unable to get process Id for %s!", processName);
            s_logger.error(msg);
            throw new ConfigurationException(msg, e);
        }
```
or
```suggestion
        String processName;
        try {
            processName = ManagementFactory.getRuntimeMXBean().getName();
            _pid = Integer.parseInt(processName.split("@")[0]);
        } catch (Exception e) {
            String msg = String.format("Unable to get process Id for %s!", processName);
            s_logger.debug(msg , e);
            throw new ConfigurationException(msg, e);
        }
```